### DISCUSSION TITLE
Placeholder not disappearing after page load

### DISCUSSION THREAD

**Discussion by SalazarJosh** [gatsbyjs/gatsby#39114] [Help]
I have a GatsbyImage component with the default `object-fit: cover;' behavior, but it's not being recognized due to the placeholder not disappearing. I'm also making these assumptions:
1. The placeholder is supposed to disappear
2. The placeholder is the first child `<div>` in the resulting <GatsbyImage> HTML

Here's the resulting HTML from the `<GatsbyImage>` component

```
<div data-gatsby-image-wrapper="" class="gatsby-image-wrapper gatsby-image-wrapper-constrained">
    /* IS THIS THE PLACEHOLER ??? */
    <div style="max-width:920px;display:block">
        <img alt="" role="presentation" aria-hidden="true" ...>
    </div>
    <div aria-hidden="true" data-placeholder-image="" style="opacity: 0; transition: opacity 500ms linear; object-fit: cover;"></div>
    <picture>
        <source .... />
        <img .... />
    </picture>
    </noscript>
</div>
```
Breaking this down, the first child `<div>` element has an `<img>` child that has `role="presentation"` and `aria-hidden="true"`. I'm assuming this is the placeholder because of that and because the `<img>` is an empty svg. The resulting image, after load, should be the `<img>` under the `<picture>`.

If this is the case, the placeholder `<div>` should disappear, otherwise the `object-fit: cover;` and any other position styling doesn't work because the placeholder has a `display:block;`. After the page renders, if I delete the placeholder `<div>` the image scales properly to respect the `object-fit: cover;` property.

In short, I'm trying to get an image to fill the available space _without_ expanding the container. With the placeholder and the `display:block` on it, the container will always be the size of the placeholder. Since we don't have much control over the styling of the placeholder, I cannot get the `object-fit: cover;` to work as expected.

Am I overlooking something, or is this a bug? The [Gatsby homepage](https://www.gatsbyjs.com/) has similar functionality for the images (placeholders stick around with `display: block;` on them).

**Comment by nop33** [comment]
I noticed the same issue, only in production builds.

**Comment by ranjanashwin** [comment]
You're correct in identifying that the <div> with data-placeholder-image is the placeholder rendered by the GatsbyImage component. It's intended to fade out once the actual image loads, but in some cases—particularly in production builds—it remains in the DOM with opacity: 0 and display: block, rather than being removed entirely. This can interfere with layout behavior, especially when you're using styles like object-fit: cover, as the placeholder div may continue to occupy space and affect how the container resizes. One effective approach is to use layout="fill" on the GatsbyImage component, combined with style={{ position: 'absolute' }} and imgStyle={{ objectFit: 'cover' }}, while ensuring the parent container has position: relative and overflow: hidden. For example: <GatsbyImage image={image} alt="" style={{ position: 'absolute' }} imgStyle={{ objectFit: 'cover' }} layout="fill" />. If the placeholder isn't necessary for your use case, you can also disable it entirely using placeholder="none" like this: <GatsbyImage image={image} alt="" placeholder="none" />, which helps eliminate any impact on layout. Lastly, be sure to run gatsby clean && gatsby build && gatsby serve to test this behavior in a clean production environment, as lingering cache can also cause inconsistencies. Let me know if you'd like help reviewing a minimal reproduction—happy to assist.
